pkg/tools/logging/query.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/template"
+	"time"
+
+	logging "cloud.google.com/go/logging/apiv2"
+	"cloud.google.com/go/logging/apiv2/loggingpb"
+	"github.com/GoogleCloudPlatform/gke-mcp/pkg/config"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	_ "google.golang.org/genproto/googleapis/cloud/audit" // Import for AuditLog proto so we can convert to JSON.
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type LogQueryRequest struct {
+	Query     string     `json:"query"`
+	ProjectID string     `json:"project_id"`
+	TimeRange *TimeRange `json:"time_range,omitempty"`
+	Since     string     `json:"since,omitempty"`
+	Limit     int        `json:"limit,omitempty"`
+	Format    string     `json:"format,omitempty"`
+}
+
+type TimeRange struct {
```

Comment:
I don't think this needs to be public / exported.

Reply:
Same here.